src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,334 @@
+                            byte[] bufferClone = (byte[])buffer.Clone();
+                            ConvertToHexidecimal(bufferClone, path, offset);
+                        }
+                        else
+                        {
+                            // Handle the case of a partial (and probably last) buffer.  Copies the bytes read into a new,
+                            // shorter array so we do not have the extra bytes from the previous pass through at the end.
+                            byte[] remainingBytes = new byte[bytesRead];
+                            Array.Copy(buffer, 0, remainingBytes, 0, bytesRead);
+                            ConvertToHexidecimal(remainingBytes, path, offset);
+                        }
+                        // Update offset value.
+                        offset += (UInt32)bytesRead;
+                    }
+                }
+            }
+            catch (IOException ioException)
+            {
+                // IOException takes care of FileNotFoundException, DirectoryNotFoundException, and PathTooLongException
+                WriteError(new ErrorRecord(ioException, "FormatHexIOError", ErrorCategory.WriteError, path));
+            }
+            catch (ArgumentException argException)
+            {
+                WriteError(new ErrorRecord(argException, "FormatHexArgumentError", ErrorCategory.WriteError, path));
+            }
+            catch (NotSupportedException notSupportedException)
+            {
+                WriteError(new ErrorRecord(notSupportedException, "FormatHexPathRefersToANonFileDevice", ErrorCategory.InvalidArgument, path));
+            }
+            catch (SecurityException securityException)
+            {
+                WriteError(new ErrorRecord(securityException, "FormatHexUnauthorizedAccessError", ErrorCategory.PermissionDenied, path));
+            }
+        }
+
+        #endregion
+
+        #region InputObjects
+
+        /// <summary>
+        /// Creates a byte array from the object passed to the cmdlet (based on type) and passes
+        /// that array on to the ConvertToHexidecimal method to output.
+        /// </summary>
+        /// <param name="inputObject"></param>
+        private void ProcessObjectContent(PSObject inputObject)
+        {
+            Object obj = inputObject.BaseObject;
+            byte[] inputBytes = null;
+            if (obj is System.IO.FileSystemInfo)
+            {
+                string[] path = { ((FileSystemInfo)obj).FullName };
+                List<string> pathsToProcess = ResolvePaths(path, true);
+                ProcessPath(pathsToProcess);
+            }
+            
+            else if (obj is string)
+            {
+                string inputString = obj.ToString();
+                Encoding resolvedEncoding = EncodingConversion.Convert(this, Encoding);
+                inputBytes = resolvedEncoding.GetBytes(inputString);
+            }
+            
+            else if (obj is byte)
```

Comment:
It would be nice to see all primitive types (including enums) and arrays of these types - instead of this random assortment.

I wrote some code that covers much of what I think is useful and it's a similar number of lines of code, see https://gist.github.com/lzybkr/9940b63c8301c31316bdb3ec6305536f